### PR TITLE
Add Pnpm support (follow-up of #162)

### PR DESCRIPTION
**Pull request by cezaraugusto**
Follow-up PR of https://github.com/extension-js/extension.js/pull/162 after feedback reported in https://github.com/extension-js/extension.js/pull/162#issuecomment-2325080407.

Screenshot 2024-09-03 at 00 20 34

### CODE REVIEW

**Comment by cezaraugusto** [comment]
@karlhorky can you please confirm if v2.0.0-alpha.18 works? that's what i run in the screenshot above

```
pnpx extension@2.0.0-alpha.18 create my-extension --template content-react
```

**Comment by karlhorky** [comment]
@cezaraugusto if you can, make sure to also try `pnpm tsc` (to run the TypeScript compiler)

**Comment by cezaraugusto** [comment]
@karlhorky thanks for figuring out the issues! I went over your comment and recorded a video with a working sample. When you get a chance please let me know if it works for you too!

```
pnpx extension create my-extension --template=content-react
```

issue147

**Comment by karlhorky** [comment]
Nice!! Video looks mostly good, I noticed 2 things:

A) There was no testing of [using the `chrome` global in `background.ts`](https://github.com/extension-js/extension.js/pull/162#issuecomment-2325080407:~:text=adding%20a%20usage%20of%20the%20chrome%20global%20(from%20%40types/chrome)%20to%20background.ts%20then%20fails)

**Comment by cezaraugusto** [comment]
@karlhorky sure! made a sample extension here https://github.com/cezaraugusto/sample-background-action-type. Below is the recording. 

https://github.com/user-attachments/assets/4214d066-d476-46c5-9df6-e864d6038189

cc @OSpoon if you get a chance to review this one as well

**Comment by karlhorky** [comment]
@cezaraugusto great, thanks! Added a GitHub Actions CI workflow to run `pnpm tsc` here:

- https://github.com/cezaraugusto/sample-background-action-type/pull/1

File: programs/create/steps/generate-extension-types.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export async function generateExtensionTypes(
   const typePath =
     process.env.EXTENSION_ENV === 'development'
       ? path.resolve(process.cwd(), 'programs/cli/types')
-      : 'extension/dist/types'
+      : './node_modules/extension/dist/types'
```
Comment (karlhorky) [inline programs/create/steps/generate-extension-types.ts:20]: does reverting this change work?

```suggestion
      : 'extension/dist/types'
```

that was part B of [my comment](https://github.com/extension-js/extension.js/pull/165#issuecomment-2330860718):

> B) in the `extension-env.d.ts`, there are paths to `./node_modules/extension/...` - I think you can use `extension/...` with those, without `./node_modules`

File: programs/develop/commands/commands-lib/generate-extension-types.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export async function generateExtensionTypes(projectPath: string) {
   const typePath =
     process.env.EXTENSION_ENV === 'development'
       ? path.resolve(process.cwd(), 'programs/cli/types')
-      : 'extension/dist/types'
+      : './node_modules/extension/dist/types'
```
Comment (karlhorky) [inline programs/develop/commands/commands-lib/generate-extension-types.ts:17]: does reverting this change work?

```suggestion
      : 'extension/dist/types'
```

that was part B of [my comment](https://github.com/extension-js/extension.js/pull/165#issuecomment-2330860718):

> B) in the `extension-env.d.ts`, there are paths to `./node_modules/extension/...` - I think you can use `extension/...` with those, without `./node_modules`

Comment (cezaraugusto) [reply programs/develop/commands/commands-lib/generate-extension-types.ts]: works indeed! should be live in v2.0.0-alpha.23

File: programs/create/steps/generate-extension-types.ts (hunk 1 above)
Comment (cezaraugusto) [reply programs/create/steps/generate-extension-types.ts]: works indeed! should be live in v2.0.0-alpha.23

**Comment by karlhorky** [comment]
Thanks to you too! Great job 🙌

**Pull request merged**